File: src/main/java/dev/slne/spawn/trader/command/subcommand/SpawnTraderClearCommand.java
package dev.slne.spawn.trader.command.subcommand;

import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.spawn.trader.SpawnTrader;
import dev.slne.spawn.trader.entity.impl.TraderNPC;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;

/**
 * The type Spawn trader clear command.
 */
public class SpawnTraderClearCommand extends CommandAPICommand {

  private final SpawnTrader instance = SpawnTrader.instance();
  private final TraderNPC traderNPC = this.instance.traderNPC();

  /**
   * Instantiates a new Spawn trader clear command.
   *
   * @param name the name
   */
  public SpawnTraderClearCommand(String name) {
    super(name);

    executesPlayer((player, args) -> {

      if(traderNPC.clear("spawn-trader")) {
        player.sendMessage(SpawnTrader.prefix().append(Component.text("Der Trader wurde entfernt.").color(NamedTextColor.GREEN)));
      } else {
        player.sendMessage(SpawnTrader.prefix().append(Component.text("Der Trader existiert nicht, oder ein ZNPCsPlus Fehler ist aufgetreten.").color(NamedTextColor.RED)));
      }
    });
  }
}

File: src/main/java/dev/slne/spawn/trader/command/subcommand/SpawnTraderClearCommand.kt
package dev.slne.spawn.trader.command.subcommand

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.spawn.trader.plugin
import dev.slne.spawn.trader.util.TraderPermissionRegistry
import dev.slne.surf.surfapi.core.api.messages.adventure.sendText

class SpawnTraderClearCommand(name: String) : CommandAPICommand(name) {
    private val traderNPC = plugin.traderNPC

    init {
        withPermission(TraderPermissionRegistry.COMMAND_CLEAR)
        playerExecutor { player, _ ->
            if (traderNPC.clear("spawn-trader")) {
                player.sendText {
                    appendPrefix()
                    success("Der Trader wurde entfernt.")
                }
            } else {
                player.sendText {
                    appendPrefix()
                    error("Der Trader existiert nicht, oder ein surf-npc Fehler ist aufgetreten.")
                }
            }
        }

    }
}
